theano/compile/function_module.py

```diff
@@ -856,7 +863,7 @@ def __call__(self, *args, **kwargs):
         # Do the actual work
         t0_fn = time.time()
         try:
-            outputs = self.fn()
+            outputs = self.fn(self.output_idx)
```

Comment:
Done, although as far as I see it executes [vm.__call__()](https://github.com/sygi/Theano/blob/partial-evaluation/theano/gof/vm.py#L397), which didn't have any arguments before.